ships/0026-image-abstraction-layer.md

```diff
@@ -0,0 +1,232 @@
+| from        | Indicates the source of the image (from where the operator should import it), if not set the operator will use the spec.targetImage.From by default. (optional) |
+| mirror      | Informs if the Image should be mirrored in the configured mirror registry. If not set targetImage.Mirror is used instead. (optional) |
+| insecure    | Indicates if the remote registry hosting the image is using “invalid certificates”, i.e. skip TLS verification. If not set spec.targetImage.Insecure is used instead. (optional) |
+
+If an ImageImport has in its `spec.targetImage` an Image that does not exist yet one should be
+created by the operator before starting to process the ImageImport object.
+
+### Mirroring images
+
+If mirroring is set in an Image or ImageImport the operator will attempt to mirror the image
+content into another registry provided by the user. To mirror images locally one needs to inform
+the operator about the mirror registry location and authentication methods. Such mirror registry
+could be configured through a Secret called `mirror-registry-config`, this secret would contain
+the following properties:
+
+| Name       | Description                                                                       |
+| -----------| --------------------------------------------------------------------------------- |
+| address    | The mirror registry URL                                                           |
+| username   | Username                                                                          |
+| password   | The password to be used                                                           |
+| token      | The auth token to be used                                                         |
+| insecure   | Allows access to insecure registries if set to "true"                             |
+| repository | If set all images are going to be mirrored inside this registry repository        |
+
+### Image status
+
+The status field for a given Image should contain all its previously imported versions of the
+same image sorted by import time. Follow below the proposed properties found in an Image `.status`
+property and their meaning:
+
+| Name              | Description                                                                |
+| ----------------- | -------------------------------------------------------------------------- |
+| hashReferences    | A list of all imported versions of the Image. Sorted by import time descending (the first item in this list is the last imported version) |
+   
+The property `.status.hashReferences` is a list of imported image versions, The operator would
+hold up to twenty five imports per image. Each item in the list is composed of the following
+properties:
+
+| Name           | Description  |
+| -------------- | ------------ |
+| from           | Keeps a reference from where the image got imported |
```

Comment:
Will this include the digest?

Reply:
No, what contains the digest is the `imageReference` property below. As the Image may have its "From" property changed in time it is useful, IMHO, to have recorded what the "From" was the moment of import. WDYT?